projects/step-frontend/src/lib/modules/timeseries/pages/dashboard/dashboard-page.component.ts

```diff
@@ -0,0 +1,306 @@
+  selected: boolean;
+}
+
+interface PageParams {
+  editMode?: boolean;
+}
+
+const EDIT_PARAM_NAME = 'edit';
+
+@Component({
+  selector: 'step-dashboard-page',
+  templateUrl: './dashboard-page.component.html',
+  styleUrls: ['./dashboard-page.component.scss'],
+})
+export class DashboardPageComponent implements OnInit, OnDestroy {
+  readonly AGGREGATES: AggregationType[] = ['SUM', 'AVG', 'MIN', 'MAX', 'COUNT', 'RATE', 'MEDIAN'];
+  readonly DASHLET_HEIGHT = 300;
+
+  @ViewChildren(ChartDashletComponent) dashlets: Dashlet[] = [];
+  @ViewChild(DashboardFilterBarComponent) filterBar?: DashboardFilterBarComponent;
+
+  private _timeSeriesService = inject(TimeSeriesService);
+  private _timeSeriesContextFactory = inject(TimeSeriesContextsFactory);
+  private _dashboardService = inject(DashboardsService);
+  private _route: ActivatedRoute = inject(ActivatedRoute);
+  private _router: Router = inject(Router);
+  private _authService: AuthService = inject(AuthService);
+  colorsPool: TimeseriesColorsPool = new TimeseriesColorsPool();
+
+  dashboard!: DashboardView;
+  dashboardBackup!: DashboardView;
+
+  context!: TimeSeriesContext;
+
+  compareModeEnabled = false;
+  timeRangeOptions: TimeRangePickerSelection[] = TimeSeriesConfig.ANALYTICS_TIME_SELECTION_OPTIONS;
+  timeRangeSelection: TimeRangePickerSelection = this.timeRangeOptions[0];
+
+  terminator$ = new Subject<void>();
+
+  editMode = false;
+  metricTypes?: MetricType[];
+
+  hasWritePermission = false;
+
+  ngOnInit(): void {
+    const pageParams = this.extractUrlParams();
+    this.removeOneTimeUrlParams();
+    this.hasWritePermission = this._authService.hasRight('dashboard_write');
+    this._route.paramMap.subscribe((params) => {
+      const id: string = params.get('id')!;
+      if (!id) {
+        throw new Error('Dashboard id not present');
+      }
+      this._dashboardService.getEntityById5(id).subscribe((dashboard) => {
+        this.dashboard = dashboard;
+        this.context = this.createContext(this.dashboard);
+        this.subscribeForContextChange();
+        this.subscribeForTimeRangeChange();
+        if (pageParams.editMode && this.hasWritePermission) {
+          this.fetchMetricTypes();
+          this.enableEditMode();
+        }
+      });
+    });
+  }
+
+  private extractUrlParams(): PageParams {
+    const initialParams: Params = this._route.snapshot.queryParams;
+    return {
+      editMode: initialParams[EDIT_PARAM_NAME] == '1',
+    };
+  }
+
+  ngOnDestroy(): void {
+    this._timeSeriesContextFactory?.destroyContext(this.context?.id);
+  }
+
+  enableEditMode() {
+    this.dashboardBackup = { ...this.dashboard, dashlets: [...this.dashboard.dashlets.map((item) => ({ ...item }))] };
+    this.editMode = true;
+    if (!this.metricTypes) {
+      this.fetchMetricTypes();
+    }
+  }
+
+  private fetchMetricTypes() {
+    this._timeSeriesService.getMetricTypes().subscribe((metrics) => (this.metricTypes = metrics));
+  }
+
+  disableEditMode() {
```

Comment:
I would call it "cancelEditMode" since save also disavlesEditMode but saves changes.

Reply:
agree